portal-impl/src/com/liferay/portal/action/UpdatePasswordAction.java

```diff
@@ -302,9 +305,31 @@ else if (authType.equals(CompanyConstants.AUTH_TYPE_ID)) {
 			login = String.valueOf(userId);
 		}
 
-		AuthenticatedSessionManagerUtil.login(
-			httpServletRequest, httpServletResponse, login, password1, false,
-			null);
+		try {
+			boolean currentValidate = isValidatePassword(httpServletRequest);
+
+			PwdToolkitUtilThreadLocal.setValidate(currentValidate);
+
+			User userUpdated = UserLocalServiceUtil.updatePassword(
+				userId, password1, password2, passwordReset);
```

Comment:
For e-mail sending the already existing `user` object would be sufficient enough in my opinion.